tasks/contactdefinitive/app/src/main/java/com/example/contactdefinitive/ResultActivity.kt

```diff
@@ -0,0 +1,247 @@
+                val resultValue = String.format("%.2f", resultado)
+                val completeValue = "$resultValue $coinValue2"
+                coin1.text = "$valor $coinValue1"
+                coinResult.text = completeValue
+                imgCHF1.visibility = View.VISIBLE
+                imgEUR2.visibility = View.VISIBLE
+            }
+
+            (coinValue1 == "EUR" && coinValue2 == "BRL" && valor != 0f) -> with(binding) {
+                val resultado = (valor!! * 5.72f)
+                val resultValue = String.format("%.2f", resultado)
+                val completeValue = "$resultValue $coinValue2"
+                coin1.text = "$valor $coinValue1"
+                coinResult.text = completeValue
+                imgEUR1.visibility = View.VISIBLE
+                imgBR2.visibility = View.VISIBLE
+            }
+
+            (coinValue1 == "EUR" && coinValue2 == "USD" && valor != 0f) -> with(binding) {
+                val resultado = (valor!! * 0.42f)
+                val resultValue = String.format("%.2f", resultado)
+                val completeValue = "$resultValue $coinValue2"
+                coin1.text = "$valor $coinValue1"
+                coinResult.text = completeValue
+                imgEUR1.visibility = View.VISIBLE
+                imgUSD2.visibility = View.VISIBLE
+            }
+
+            (coinValue1 == "EUR" && coinValue2 == "GBP" && valor != 0f) -> with(binding) {
+                val resultado = (valor!! / 1.02f)
+                val resultValue = String.format("%.2f", resultado)
+                val completeValue = "$resultValue $coinValue2"
+                coin1.text = "$valor $coinValue1"
+                coinResult.text = completeValue
+                imgEUR1.visibility = View.VISIBLE
+                imgGBP2.visibility = View.VISIBLE
+            }
+
+            (coinValue1 == "EUR" && coinValue2 == "CHF" && valor != 0f) -> with(binding) {
+                val resultado = (valor!! / 0.19f)
+                val resultValue = String.format("%.2f", resultado)
+                val completeValue = "$resultValue $coinValue2"
+                coin1.text = "$valor $coinValue1"
+                coinResult.text = completeValue
+                imgEUR1.visibility = View.VISIBLE
+                imgCHF2.visibility = View.VISIBLE
+            }
+
+            (coinValue1 == "Selecionar" || coinValue2 == "Selecionar") -> with(binding) {
+                Toast.makeText(this@ResultActivity, R.string.selected_close_toast, Toast.LENGTH_SHORT).show()
+                finish()
+            }
+
+            (coinValue1 == coinValue2) -> with(binding) {
+                Toast.makeText(this@ResultActivity, R.string.required_distint_coins_toast, Toast.LENGTH_SHORT).show()
+                finish()
+            }
+
+            else -> with(binding) {
+                Toast.makeText(this@ResultActivity, R.string.required_value_toast, Toast.LENGTH_SHORT).show()
+                finish()
+            }
+        }
+
```

Comment:
remove it from setupListeners and move to setupView and use a "Map" to avoid all of these conditions

Reply:
when you pu a too high value, it breaks the content. adjust it 
![image](https://github.com/user-attachments/assets/1abe34d3-19b3-4d18-ac3c-793c161cca25)

Also the flags aren't with the same size values
![image](https://github.com/user-attachments/assets/178131c1-2d40-498b-8f61-ef1ba4c0af3f)